### PR TITLE
GH-3897: Deprecate `ChannelSecurityInterceptor`

### LINKED ISSUE #3897: Revise Spring Integration security in favor of Spring Security Messaging
The `AbstractSecurityInterceptor` (and supporting API) is deprecated in Spring Security now.
Consider to deprecate `ChannelSecurityInterceptor` in favor of `AuthorizationChannelInterceptor` from Spring Security Messaging.
With that move we probably will deprecate a `@SecuredChannel` and all the AOP infrastructure we have in `spring-integration-security`.
The fix depends on the https://github.com/spring-projects/spring-security/issues/11915 which will bring for us a `postReceive()` security contract.
The current `receive()` AOP interceptor does not make too much sense since there might be just no a static `SecurityContext` to auth against.
The `postReceive()` phase may be already populated from the received message via `SecurityContextPropagationChannelInterceptor.postReceive()` where an `Authentication` can be propagated from the producer side.

Also: the migration to the `AuthorizationChannelInterceptor` and its infrastructure we will have a long time ago requested SpEL support for security: https://jira.spring.io/projects/INT/issues/INT-1479

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-projects/spring-integration/issues/3897

Spring Security has deprecated `AccessDecisionManager` and all its infrastructure in favor of `AuthorizationManager`

* Deprecate and AOP `ChannelSecurityInterceptor` and all its infrastructure, including `@SecuredChannel` and respective XML configuration. The `AuthorizationChannelInterceptor` added to respective channels for security or configured as a global channel interceptor fully covers the previous AOP configuration
* Fix deprecation warnings in other tests with security

### CODE REVIEW

File: src/reference/asciidoc/security.adoc [hunk 1]
```diff
@@ -7,6 +7,10 @@ Messaging independence and loose coupling let target systems communicate with ea
 We can either trust all those messages or secure our service against "`infecting`" messages.
 
 Spring Integration, together with https://projects.spring.io/spring-security/[Spring Security], provides a simple and comprehensive way to secure message channels, as well as other part of the integration solution.
+Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of `AuthorizationChannelInterceptor` from `spring-security-messaging` module.
```
Comment (garyrussell) [inline src/reference/asciidoc/security.adoc:10]:
```suggestion
Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of using `AuthorizationChannelInterceptor` from the `spring-security-messaging` module.
```

File: src/reference/asciidoc/security.adoc [hunk 2]
```diff
@@ -7,6 +7,10 @@ Messaging independence and loose coupling let target systems communicate with ea
...
+The respective `AuthorizationManager` infrastructure fully covers previously supported role-based authentication, plus it allows to configure any other possible authorization strategies including.
```
Comment (garyrussell) [inline src/reference/asciidoc/security.adoc:11]:
```suggestion
The respective `AuthorizationManager` infrastructure fully covers the previously supported role-based authentication, plus it allows the configuration of any other possible authorization strategies.
```

**Review by garyrussell — CHANGES_REQUESTED** [review]
Just doc polishing.

**Review by garyrussell — APPROVED** [review]

**Comment by garyrussell** [comment]
Needs rebase/conflicts.

**Comment by artembilan** [comment]
Looks like there are more deprecation warnings.

Do not merge yet.

**Comment by garyrussell** [comment]
>221 tests completed, 78 failed, 1 skipped

**Comment by artembilan** [comment]
@garyrussell ,

Looks like that one was an interim failure.
It is good now.
I guess the problem was related to the release process we have these days.

**Comment by garyrussell** [comment]
Conflict in build.grade.

**Pull request merged**